packages/inference/src/tasks/nlp/textGenerationStream.ts

```diff
@@ -85,7 +86,7 @@ export interface TextGenerationStreamOutput {
  * Use to continue text from a prompt. Same as `textGeneration` but returns generator that can be read one token at a time
  */
 export async function* textGenerationStream(
-	args: TextGenerationArgs,
+	args: BaseArgs & TextGenerationInput,
 	options?: Options
 ): AsyncGenerator<TextGenerationStreamOutput> {
```

Comment:
shoudld `TextGenerationStreamOutput` live in `@huggingface/tasks/src/tasks/text-generation/inference` just like `TextGenerationInput` & `TextGenerationOutput` ?

Reply:
Good question - cc @Wauplin with whom we discussed that previously

The current philosophy is to not type the streaming mode because it's transfer-specific, not inference-specific

Reply:
Indeed, we mentioned it here once: https://github.com/huggingface/huggingface.js/pull/468#discussion_r1484448923.

As a matter of fact, I came to the conclusion today that we should specify the `stream` parameter and the streamed output in our JS specs. I am currently starting to use the generated types in Python (see [ongoing PR](https://github.com/huggingface/huggingface_hub/pull/2036)) and for now I've kept `text_generation` apart since I'm missing `TextGenerationStreamResponse` (defined [here](https://github.com/huggingface/huggingface_hub/blob/e05fb737a08bfa5c6b9b40998b77c0ef7548a8f8/src/huggingface_hub/inference/_text_generation.py#L445) but I don't want to mix the auto-generated types with previous definitions). Agree it's more "transfer-specific" rather than "inference-specific" but the thing is that setting `stream=True` is modifying the output format so we need to document that somewhere.